code/game/objects/items/robot/cyborg_gripper.dm

```diff
@@ -169,10 +150,138 @@
+			playsound(loc, 'sound/weapons/tap.ogg', 50, TRUE, -1)
+			return
+
+		if(ishuman(M))
+			user.visible_message(
+				"<span class='notice'>[user] hugs [M] to make [M.p_them()] feel better.</span>",
+				"<span class='notice'>You hug [M] to make [M.p_them()] feel better.</span>"
+				)
+		else
+			user.visible_message(
+				"<span class='notice'>[user] pets [M]!</span>",
+				"<span class='notice'>You pet [M]!</span>"
+				)
+		playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+		return
+
+	if(user.a_intent == INTENT_HARM && !emagged)
+		if(M == user)
+			user.visible_message(
+			"<span class='notice'>[user] gives [user.p_themselves()] a firm bear-hug to make [user.p_themselves()] feel better.</span>",
+			"<span class='notice'>You give yourself a firm bear-hug to make yourself feel better.</span>"
+			)
+			playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+			return
+
+		if(!ishuman(M) || user.zone_selected == BODY_ZONE_HEAD)
+			user.visible_message(
+				"<span class='warning'>[user] bops [M] on the head!</span>",
+				"<span class='warning'>You bop [M] on the head!</span>"
+				)
+			user.do_attack_animation(M, ATTACK_EFFECT_PUNCH)
+			playsound(loc, 'sound/weapons/tap.ogg', 50, TRUE, -1)
+		else
+			user.visible_message(
+				"<span class='warning'>[user] hugs [M] in a firm bear-hug! [M] looks uncomfortable...</span>",
+				"<span class='warning'>You hug [M] firmly to make [M.p_them()] feel better! [M] looks uncomfortable...</span>"
+				)
+			playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+		return
+
+	// Just in case.
+	if(!emagged)
+		return
+
+	if(M == user)
+		user.visible_message(
+			"<span class='danger'>[user] punches [user.p_themselves()] in the face!.</span>",
+			"<span class='userdanger'>You punch yourself in the face!</span>"
+			)
+		user.do_attack_animation(M, ATTACK_EFFECT_PUNCH)
+		playsound(loc, 'sound/weapons/smash.ogg', 50, TRUE, -1)
+		user.adjustBruteLoss(15)
+		return
+
+	if(ishuman(M))
+		// Try to punch them in the face... Unless it fell off or something.
+		if(user.zone_selected == BODY_ZONE_HEAD && M.get_organ("head"))
+			user.visible_message(
+				"<span class='danger'>[user] punches [M] squarely in the face!</span>",
+				"<span class='danger'>You punch [M] in the face!</span>"
+				)
+			var/obj/item/organ/external/head/their_face = M.get_organ("head")
+			user.do_attack_animation(M, ATTACK_EFFECT_PUNCH)
+			playsound(loc, 'sound/weapons/smash.ogg', 50, TRUE, -1)
+			their_face.receive_damage(15)
+			M.UpdateDamageIcon()
+			return
+
+	user.visible_message(
+		"<span class='danger'>[user] crushes [M] in [user.p_their()] grip!</span>",
+		"<span class='danger'>You crush [M] in your grip!</span>"
+		)
+	playsound(loc, 'sound/weapons/smash.ogg', 50, TRUE, -1)
+	M.adjustBruteLoss(15)
+
+// MARK: Gripper Types
+
+/* Universal Gripper
+* Not supplied to any cyborg by default. 
+* Can be varedited onto a borg for event stuff. 
+* Functions almost like a real hand!
+*/
```

Comment:
```suggestion
/* 
*Universal Gripper
* Not supplied to any cyborg by default. 
* Can be varedited onto a borg for event stuff. 
* Functions almost like a real hand!
*/
```
Nit of myself, since I requested some changes on this earlier.